components/playlist/browser/resources/components/header.tsx

```diff
@@ -120,29 +145,90 @@ export default function Header ({ playlistId }: HeaderProps) {
   const isDefaultPlaylist = playlist?.id === 'default'
   if (contextualMenuItems && !isDefaultPlaylist) {
     contextualMenuItems.unshift({
-      name: 'Rename',
+      name: getLocalizedString('bravePlaylistContextMenuRenamePlaylist'),
       iconName: 'edit-box',
-      onClick: () => {}
+      onClick: () => {
+        getAllActions().setPlaylistEditMode(PlaylistEditMode.RENAME)
+      }
     })
     contextualMenuItems.push({
-      name: 'Delete playlist',
+      name: getLocalizedString('bravePlaylistContextMenuDeletePlaylist'),
       iconName: 'trash',
       onClick: () => {
-        getPlaylistAPI().showRemovePlaylistUI(playlistId!)
+        getPlaylistAPI().showRemovePlaylistUI(playlistId)
       }
     })
   }
 
   const totalDuration = useTotalDuration(playlist)
   const totalSize = useTotalSize(playlist)
+  const playlistEditMode = usePlaylistEditMode()
 
-  return (
-    <HeaderContainer>
-      {playlist ? (
+  const [hasNewName, setHasNewName] = React.useState(true)
+  const inputElementRef = React.useRef<HTMLInputElement>(null)
+
+  const onSave = () => {
+    if (
+      !playlist ||
+      playlist?.name === inputElementRef.current?.value ||
+      hasNewName
+    ) {
+      return
+    }
+
+    getPlaylistAPI().renamePlaylist(
+      playlist.id!,
+      inputElementRef.current!.value
+    )
+    getAllActions().setPlaylistEditMode(undefined)
+  }
+
+  React.useEffect(() => {
+    // on unmount, resets 'edit mode'.
+    return () => {
+      getAllActions().setPlaylistEditMode(undefined)
+    }
+  }, [])
+
+  return !playlist ? (
```

Comment:
Optional: You can just return `null` instead of an empty fragment

Reply:
Done.